### PR TITLE
replace kernel-headers package with kernel-devel package for mariner 2.0 cse script

### PR DESCRIPTION
**Pull request by SeanDougherty**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:
The kernel devel package is needed by the Defender team to successfully support Defender on Mariner+AKS. Kernel-headers was previously the package thought to provide the support .

**Which issue(s) this PR fixes**:

Fixes WorkItem [#41265898](https://microsoft.visualstudio.com/OS/_sprints/taskboard/Mariner%20Platform/OS/2210?workitem=41265898) and WorkItem [#41265919](https://microsoft.visualstudio.com/OS/_sprints/taskboard/Mariner%20Platform/OS/2210?workitem=41265919)

**Requirements**:
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [x] includes documentation
- [x] adds unit tests
- [x] tested upgrade from previous version

**Special notes for your reviewer**:
Testing:

Was able to create a new AKS customized Mariner 2.0 image with a local AgentBaker environment. AKS clusters generated with this image (using --aks-custom-headers) were successfully able to call az aks update --enable-defender. Individuals from the Defender team were able to verify that they received eBFP content on their backend from clusters with this feature enabled.

**Release note**:
```
none
```

### CODE REVIEW

**Review by orparnes — APPROVED** [review]

**Review by alexeldeib — APPROVED** [review]

**Comment by alexeldeib** [comment]
missing a `make` to update generated files but lgtm after that

**Review by christopherco — APPROVED** [review]
I recommend updating the PR title and make sure final commit message reflect that this is replacing `kernel-headers` with `kernel-devel`, instead of adding `kernel-devel`.

**Pull request merged**
